File: bootstrap/v0.0.8.5/benchmarks/python/binarysearch.py
def main():
    n = 10000
    searches = 1000
    arr = list(range(n))

    total = 0
    for i in range(searches):
        target = i % n
        left = 0
        right = n - 1

        while left <= right:
            mid = (left + right) // 2
            if arr[mid] == target:
                total += mid
                break
            elif arr[mid] < target:
                left = mid + 1
            else:
                right = mid - 1

    return 0

if __name__ == "__main__":
    exit(main())

File: bootstrap/archived_runa_0.0.7.5+/v0.0.8.4/benchmarks/java/Binarysearch.java
public class Binarysearch {
    public static void main(String[] args) {
        int n = 10000;
        int searches = 1000;
        int[] arr = new int[n];

        for (int i = 0; i < n; i++) {
            arr[i] = i;
        }

        int total = 0;
        for (int i = 0; i < searches; i++) {
            int target = i % n;
            int left = 0;
            int right = n - 1;

            while (left <= right) {
                int mid = (left + right) / 2;
                if (arr[mid] == target) {
                    total += mid;
                    break;
                } else if (arr[mid] < target) {
                    left = mid + 1;
                } else {
                    right = mid - 1;
                }
            }
        }

        System.exit(0);
    }
}
